src/main/java/com/carrotcreative/cream/util/HashingUtil.java

```diff
@@ -0,0 +1,52 @@
+package com.carrotcreative.cream.util;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import android.support.annotation.NonNull;
+
+public class HashingUtil
+{
+    private static final int CACHE_SIZE = 50;
```

Comment:
Gotcha!  Okay, I think the example in the documentation is super misleading.

Looks good then :100: